gr-digital/lib/glfsr_source_f_impl.cc

```diff
@@ -21,31 +21,30 @@ namespace gr {
 namespace digital {
 
 glfsr_source_f::sptr
-glfsr_source_f::make(unsigned int degree, bool repeat, uint32_t mask, uint32_t seed)
+glfsr_source_f::make(unsigned int degree, bool repeat, uint64_t mask, uint64_t seed)
 {
     return gnuradio::make_block_sptr<glfsr_source_f_impl>(degree, repeat, mask, seed);
 }
 
 glfsr_source_f_impl::glfsr_source_f_impl(unsigned int degree,
                                          bool repeat,
-                                         uint32_t mask,
-                                         uint32_t seed)
+                                         uint64_t mask,
+                                         uint64_t seed)
     : sync_block("glfsr_source_f",
                  io_signature::make(0, 0, 0),
                  io_signature::make(1, 1, sizeof(float))),
       d_glfsr(mask ? mask : glfsr::glfsr_mask(degree), seed),
       d_repeat(repeat),
       d_index(0),
-      d_length((((uint32_t)1) << degree) - 1)
+      d_length((1L << degree) - 1)
```

Comment:
Use `1ULL` or better (?) `static_cast<uint64_t>()`: `long` has only 32 bit on x64 Windows (as far as I remember) and to avoid dealing with signed integers. Also consider using `d_length{…}` instead of `d_length(…)` to prevent narrowing conversion [1]

[1] https://en.cppreference.com/w/cpp/language/list_initialization#Narrowing_conversions

Reply:
Wilco, good spot